### ISSUE TITLE
@pixi/events Masked container with nested child containers doesn't bubble events.

### ISSUE THREAD

**Issue by jramstedt** [pixijs/pixijs#7915]
### Expected Behavior
Grandchild's event listeners should work as they do with non masked grandparent.

### Current Behavior
Grandchild's event listeners are not triggered if it's grandparent is masked.

### Steps to Reproduce
See running example.

Create following hierarchy:

Container with mask and interactive
-> Container without any event listeners or mask
-- -> Displayobject with eventlistnener

Test with mask enabled: Does not work.
Comment out adding mask: Does work.

### Environment

- **`pixi.js` version**: 6.1.3
- **Browser & Version**: Chrome 94
- **OS & Version**: Windows 10
- **Running Example**: https://www.pixiplayground.com/#/edit/t5WBOFp3-SBRfYvedOxe1

**Comment by kjarrio** [comment]
Seems to be working fine if you comment out these lines:
`delete PIXI.Renderer.__plugins.interaction;`
`renderer.addSystem(PIXI.EventSystem, 'events');`

Running example https://www.pixiplayground.com/#/edit/rJ34d_evdxC1k-6lZwEpB

**Comment by jramstedt** [comment]
> Seems to be working fine if you comment out these lines: `delete PIXI.Renderer.__plugins.interaction;` `renderer.addSystem(PIXI.EventSystem, 'events');`

Yes, This is a @pixi/events problem.

**Comment by jramstedt** [comment]
Any news?

**Comment by ShukantPal** [comment]
# It's not a bug, explanation

I've done some investigating and this is not a bug. In your code, you have the following scene graph structure:

```
Stage
 |
Container
 |
Container, Mask
|          |
Button     (no children)
```

This means the mask is like any other DisplayObject and it'll capture events passing it's hit-test _if one of it's ancestors is interactive_. Indeed, the mask you've added is a 75px square that's transparent and it is rendered above the red button. That's why the `hitTestRecursive` method returns a 0-length array on the mask (it passes hit testing but since it's not interactive, it isn't included in the propagation path of the ensuing pointerdown event):

<img width="1289" alt="Screen Shot 2022-04-09 at 10 07 07 PM" src="https://user-images.githubusercontent.com/22450567/162598139-9e2f2cdf-3765-4b4b-9fc0-15705736ed75.png">

# What you can do about it!

## 1 − (suboptimal) Reorder your scene graph

You can reorder the children of the grandparent container, such that the mask comes before the button. This will cause the hit-testing algorithm to prioritize the button over the mask.

However, the mask will continue to capture events that happen outside the button but inside the mask.

## 2 − (optimal) Set visible to false on the mask

You can instead set the [visible](https://api.pixijs.io/@pixi/display/PIXI/DisplayObject.html#visible) flag on the mask to false. The hit-testing algorithm will ignore that those objects.

I've updated the playground you've attached to this issue to demonstrate the fix.

<img width="420" alt="Screen Shot 2022-04-09 at 10 17 39 PM" src="https://user-images.githubusercontent.com/22450567/162598324-3381097b-354e-47c2-b271-eb9975a2a866.png">

https://www.pixiplayground.com/#/edit/t5WBOFp3-SBRfYvedOxe1

**Comment by ShukantPal** [comment]
@bigtimebuddy I'm open to putting in any documentation changes needed to make this more obvious.

**Comment by domis86** [comment]
@ShukantPal very good and clear explanation! 👍

**Comment by jramstedt** [comment]
Thanks for the explanation.
However, I would assume this to work the same way as before, otherwise migrating to new events can be laborious and error prone.

**Comment by jramstedt** [comment]
Wouldn't the "fix" just be to also check "isMask" when checking for visible?

**Comment by ShukantPal** [comment]
Oh yes, but you’d have to wait for a release

**Comment by jramstedt** [comment]
@ShukantPal I tested this with 7.1.2

Masks are not masking if visible = false.

This is still a problem.
Childs of masked containers doesn't work. They don't change pointer or get events.

**Comment by jramstedt** [comment]
I found a workaround.
Adding empty rectangle to mask's hitArea circumvents this problem.

`this.mask.hitArea = Rectangle.EMPTY`

**Comment by mariusk** [comment]
FWIW, I'm getting the same error. I'm using PIXI with a React app, where I control the renderer and game loop myself. Everything works fine until I try to listen to click or similar. It even works, the handler is called, but the log is spammed with these errors. I don't have a minimum example, but will see if I can make one. But something is definitively very broken, and/or hard to get "right" (external circumstances I suspect).

**Comment by Zyie** [comment]
Hey @jramstedt 

This should be fixed in 7.2.0. We now ignore masks when doing hit testing
